glean-core/ios/GleanTests/Metrics/DatetimeMetricTypeTests.swift

```diff
@@ -8,8 +8,19 @@ import XCTest
 // swiftlint:disable force_cast
 // REASON: Used in a test
 class DatetimeMetricTypeTests: XCTestCase {
+    var expectation: XCTestExpectation?
+
     override func setUp() {
+        expectation = setUpDummyStubAndExpectation(testCase: self, tag: "DatetimeMetricTypeTests")
         Glean.shared.resetGlean(clearStores: true)
+        waitForExpectations(timeout: 5.0) { error in
+            XCTAssertNil(error, "Test timed out waiting for upload: \(error!)")
+        }
+    }
```

Comment:
My other consideration was to make it a parameter to `resetGlean`...  something along the lines of `ignoreInitialPings: Boolean`. That way, for the handful of tests that do care about testing the pings generated when we reset Glean can still set up a stub to capture and test them explicitly.

Reply:
I was also annoyed at the amount I ended up repeating this.

Reply:
oh, yeah. A parameter might indeed be nicer

Reply:
Hmm, I may have spoken too soon. `XCTestExpectation` requires an `XCTestCase` instance, which is a little hard to pipe to `resetGlean` since it lives in the public API in Glean.swift. Let me instead try this by wrapping `resetGlean` in our test utils...

Reply:
Looks like the `resetGleanDiscardingInitialPings` method I made works, and allows me to delete a few more things from several test cases.